07.ls_object/lib/ls_file_stat.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+require 'etc'
+require 'time'
+require 'pathname'
+
+class LsFileStat
+  DISABLE = '-'
+  READ = 'r'
+  WRITE = 'w'
+  EXEC = 'x'
+  MODE_MAP = {
+    '0' => "#{DISABLE}#{DISABLE}#{DISABLE}",
+    '1' => "#{DISABLE}#{DISABLE}#{EXEC}",
+    '2' => "#{DISABLE}#{WRITE}#{DISABLE}",
+    '3' => "#{DISABLE}#{WRITE}#{EXEC}",
+    '4' => "#{READ}#{DISABLE}#{DISABLE}",
+    '5' => "#{READ}#{DISABLE}#{EXEC}",
+    '6' => "#{READ}#{WRITE}#{DISABLE}",
+    '7' => "#{READ}#{WRITE}#{EXEC}"
+  }.freeze
+
+  def initialize(file_path)
+    @stat = File.lstat file_path
+    @path = file_path
+  end
+
+  # ok
+  def name(show_link: false)
+    return "#{@path} -> #{readlink}" if @stat.symlink? && show_link
+
+    @path
+  end
+
+  def nlink
+    @stat.nlink
+  end
+
+  def size_in_ls_format
+    return "0x#{@stat.rdev_major}00000#{@stat.rdev_minor}" if @stat.blockdev? || @stat.chardev?
+
+    @stat.size.to_s
+  end
+
+  def permission
+    mode_octet = @stat.mode.to_s(8)[-3..].chars
+    convert_mode_str mode_octet
```

Comment:
```suggestion
    convert_mode_str(mode_octet)
```

関数的にメソッドを呼び出すなら`()`があった方が読みやすいかも（好みによるところも大きいですが）

Reply:
こちらの方がしっくりくるので同じような所はカッコつきにしてみました。